File: Core/Calculators/CalculatorFactory.cs
using Probability.Core.Contracts;
using Probability.Core.Models;
using System;

namespace Probability.Core.Calculators
{
    public class CalculatorFactory : ICalculatorFactory
    {
        public IProbabilityCalculator GetCalculator(string calculationType)
        => calculationType switch
        {
            CalculationType.CombinedWith => new CombinedWithProbabilityCalculator(),
            CalculationType.Either => new EitherProbabilityCalculator(),
            _ => throw new ArgumentException(message: $"Invalid calculation type: {calculationType}")
        };     
    }
}

File: UnitTests/Main/CalculatorFactory.fs
[<NUnit.Framework.Category("Calculator")>]
module ICalculatorFactory_test

open NUnit.Framework
open FsUnit

open Probability.Core.Calculators
open Probability.Core.Models

[<Test>]
let GetCalculator() =

    CalculatorFactory().GetCalculator(CalculationType.CombinedWith)
    |> should be instanceOfType<CombinedWithProbabilityCalculator>

    CalculatorFactory().GetCalculator(CalculationType.Either)
    |> should be instanceOfType<EitherProbabilityCalculator>